library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -341,6 +345,10 @@ private void executeDownload(State state) throws StopRequestException {
         state.resetBeforeExecute();
         setupDestinationFile(state);
 
+        if (originalDownloadInfo.shouldAllowTarUpdate(state.mimeType)) {
+            state.totalBytes = -1;
```

Comment:
@Dorvaryn have you discussed with @ouchadam ? I know he's been doing some changes around the state